coffee-model/coffee-model-base/src/main/java/hu/icellmobilsoft/coffee/model/base/audit/AuditProvider.java

```diff
@@ -19,89 +19,89 @@
 import jakarta.enterprise.context.Dependent;
 import jakarta.enterprise.inject.spi.Bean;
 import jakarta.enterprise.inject.spi.BeanManager;
 import jakarta.inject.Inject;
-
-import org.apache.deltaspike.core.util.metadata.AnnotationInstanceProvider;
-import org.apache.deltaspike.data.api.audit.CurrentUser;
-import org.apache.deltaspike.data.impl.audit.AuditPropertyException;
-import org.apache.deltaspike.data.impl.audit.PrePersistAuditListener;
-import org.apache.deltaspike.data.impl.audit.PreUpdateAuditListener;
-import org.apache.deltaspike.data.impl.property.Property;
-import org.apache.deltaspike.data.impl.property.query.AnnotatedPropertyCriteria;
-import org.apache.deltaspike.data.impl.property.query.PropertyQueries;
-import org.apache.deltaspike.data.impl.property.query.PropertyQuery;
-
-import hu.icellmobilsoft.coffee.model.base.annotation.CreatedBy;
-import hu.icellmobilsoft.coffee.model.base.annotation.ModifiedBy;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreUpdate;
 
 /**
- * Perist entity @CreatedBy property before persist with the value provided by @CurrentUser
+ * Persist entity @CreatedBy property before persist with the value provided by @CurrentUser
  *
  * Update entity @ModifiedBy property before update with the value provided by @CurrentUser
  *
  * @author czenczl
+ * @author zsolt.vasi
  * @since 1.0.0
  */
 @Dependent
-public class AuditProvider implements PrePersistAuditListener, PreUpdateAuditListener {
+public class AuditProvider extends AbstractProvider {
 
     @Inject
     private BeanManager manager;
 
-    /** {@inheritDoc} */
-    @Override
+    /**
+     * Persist entity @CreatedBy (@ModifiedBy) property before persist with the value provided by @CurrentUser
+     *
+     * @param entity
+     *            Object entity to persist
+     */
+    @PrePersist
     public void prePersist(Object entity) {
-        PropertyQuery<Object> createdByQuery = PropertyQueries.<Object> createQuery(entity.getClass())
-                .addCriteria(new AnnotatedPropertyCriteria(CreatedBy.class));
-        for (Property<Object> property : createdByQuery.getWritableResultList()) {
-            Object value = resolvePrincipal(entity, property);
-            setProperty(entity, property, value);
-        }
-
-        PropertyQuery<Object> modifiedByQuery = PropertyQueries.<Object> createQuery(entity.getClass())
-                .addCriteria(new AnnotatedPropertyCriteria(ModifiedBy.class));
-        for (Property<Object> property : modifiedByQuery.getWritableResultList()) {
-            ModifiedBy annotation = property.getAnnotatedElement().getAnnotation(ModifiedBy.class);
-            if (annotation.onCreate()) {
-                Object newValue = resolvePrincipal(entity, property);
-                setProperty(entity, property, newValue);
+        List<Field> allFields = getAllFields(entity.getClass());
+        for (Field field : allFields) {
+            if (field.isAnnotationPresent(CreatedBy.class)) {
```

Comment:
Ez a megoldas csak a fieldeken levo annotaciokra fog mukodni de ha valaki a getter metodusokon annotal akkor nem. Az elozo megoldas ezzel is szamolt, ezt nem szabadna nekunk sem korlatozni. JPA mindkettot engedi.

Reply:
Bovitettem, mar figyelembe veszi a getter metodusok annotacioit is.